src/metatensor/models/experimental/pet/utils/systems_to_batch_dict.py

```diff
@@ -238,7 +242,24 @@ def systems_to_batch_dict(
         S_list: torch.Tensor = torch.cat(S_list_raw)
         S_list = S_list.transpose(0, 1)
 
-        species: torch.Tensor = system.types
+        unique_index = torch.unique(torch.cat((i_list, j_list)))
+        species: torch.Tensor = system.types[unique_index]
+
+        # REMAPPING TO CONTIGUOUS INDEXING
```

Comment:
maybe add an explanation about why this is required?

Reply:
Thanks, good idea